### PR TITLE
Stop dask state change on Iris import

### PR DESCRIPTION
**Pull request by dkillick**
We have chosen to set default Iris behaviour to use only one thread when processing dask graphs initially. This was done by setting the global dask state using `dask.set_options()`. This is far from ideal as it means Iris is changing dask state; a situation that should not be. This PR updates the system for setting dask processing options so that behaviour is maintained without changing dask state.

### CODE REVIEW

**Comment by pelson** [comment]
Closes #2739.

File: lib/iris/_lazy_data.py [hunk 1]
```diff
@@ -42,17 +42,21 @@ def _iris_dask_defaults():
     all available CPUs.
 
     .. note::
+
         We only want Iris to set dask options in the case where doing so will
         not change user-specified options that have already been set.
 
     """
+    dask_opts = {}
     if 'pool' not in dask.context._globals and \
```
Comment (pelson) [inline lib/iris/_lazy_data.py]: I'd probably guard about accessing ```dask.context._globals```. They could legitimately remove this and leave iris unable to load data... 🤒

Comment (pelson) [reply lib/iris/_lazy_data.py]: (to be clear, in this context I simply mean ```getattr(dask.context, '_globals')```)

Comment (pelson) [reply lib/iris/_lazy_data.py]: Given the nature of this API (private), I'd also be tempted to add a test that asserts dask's set_opt behaviour continues to be a valid assumption in ```dask.context._globals```.

**Comment by dkillick** [comment]
@pelson I had a sneaky extra 10min, so I implemented the changes you suggested 😎

File: lib/iris/_lazy_data.py [hunk 2]
```diff
@@ -42,17 +42,23 @@ def _iris_dask_defaults():
...
-    if 'pool' not in dask.context._globals and \
-            'get' not in dask.context._globals:
-        dask.set_options(get=dget_sync)
-
-
-# Run this at import time to set dask options for Iris.
-_iris_dask_defaults()
+    dask_opts = {}
+    dask_globals = getattr(dask.context, '_globals')
+    if dask_globals is not None:
```
Comment (pelson) [inline lib/iris/_lazy_data.py]: So if dask change their ```dask.context._globals``` storage location, we let dask give us the default engine. Is that what is intended?

File: lib/iris/_lazy_data.py [hunk 3]
```diff
@@ -42,17 +42,23 @@ def _iris_dask_defaults():
...
+        if 'pool' not in dask_globals and \
+                'get' not in dask_globals:
+            dask_opts.update(get=dget_sync)
+        else:
+            # We may need to unset a previously-set default.
+            if dask_opts.get('get') is not None:
+                dask_opts = {key: value for key, value in dask_opts.items()
+                             if key != 'get'}
```
Comment (pelson) [inline lib/iris/_lazy_data.py]: I'm trying to understand why the get is not used from the dask options. I think this comes down to my not-understanding the dask globals and what they actually mean...

**Comment by pelson** [comment]
Deferring for now. #2879 removes the default change, and we can re-introduce a different default when we have a "smoking gun"/evidence of need to divert from dask's defaults.

**Pull request closed without merging**